### DISCUSSION TITLE
S3 client to use Supabase storage backend

### DISCUSSION THREAD

**Discussion by madsbuch** [supabase/supabase#7648] [Questions] [answered]
Is it possible, and in that case, what would credentials be, to let generic S3 clients use Supabase storage instead of S3?

A concrete example: We use retool. They allow us to integrate with S3 for storing assets. They allow me to point at s3 compatible storage backends. Here, I would like t point to supabase. Is that possible? And what would the credentials and base URL be?

![Screenshot from 2022-07-11 20-18-24](https://user-images.githubusercontent.com/1612807/178331375-56881a56-8bbe-4e0c-bcfe-0abeb1c720d2.png)
 
 This question would also apply to other tools like Strapi and other client libraries (Js, Elixir, Go, etc) that allows for S3 compatible backends.

**Comment by GaryAustin1** [comment]
The SB hosted storage shares a single s3 bucket among many instances for multiple users.  I forget the s3 method used to partition the bucket and do security.  I doubt a normal s3 client would work in this configuration (without access to the  main bucket).

**Reply by madsbuch** [reply]
Hi @GaryAustin1 

The question does not regard the backend Supabase storage uses, but rather Supabase' own API.

Like I can use Wasabi as a drop-in for S3, I would want to use Supabase as a drop-in for S3.

**Comment by iljapanic** [comment]
@madsbuch Hi there, did you ever find an answer to this? We would love to use a S3-compatible client to access Supbase storage, so our non-technical users can manage assets more easily.

**Reply by madsbuch** [reply]
we moved away from using the supabase object storage.

currently, I dont work on supabase anymore, but that was the right choice at the time as we also needed tight integration with CDNs etc. 

so my recommendation would be to just use an actual S3 bucket.

**Comment by madsbuch** [comment, marked as answer]
Seems like this is supported now!

https://supabase.com/blog/s3-compatible-storage
